schemas/computation.schema.tpl.json

```diff
@@ -0,0 +1,84 @@
+{
+    "_type": "https://openminds.ebrains.eu/computation/Computation",
+    "_extends": "../../core/research/protocolExecution.schema.tpl.json",
```

Comment:
Additionally, "_extends" will cause your schema (here: computation) to be extended by the content of the ref. schema (here: protocolExecution). 
Therefore, the 3 properties - "input", "output" and "protocol" - shouldn't be listed under properties here. You also don't have to add them as required because the original schema (protocolExecution) has them required already. 
However, looking at the details, I noticed that you would like to change the "_linkedTypes" for "input" and "output" here. As far as I'm aware, this is not possible like this. 
I'm unsure what you would like to achieve here. 
1. Do you want “computation” to have all properties from protocolExecution and the additional ones stated specifically here? Then “computation” would be a more detailed version of a “protocolExecution”. 
OR
2. Do you want to express that this “computation” is related to a specific “protocolExecution”? Then the “protocolExecution” would exist first, and the “computation” gives additional information related to the execution. 
I’m asking to figure out how the schema would need to be technically changed to express the correct alternative.   
For 1.: “input”, “output” and “protocol” are already existing properties for “protocolExecution”. These are required there, so that you could remove them from the required field for “computation”. The properties wouldn’t need to be listed under “properties” again and the original content cannot be overwritten. You used other “_linkedTypes” for “input” and “output” for “computation” than “protocolExecution”. As far as I know that’s not possible/allowed. This would need to be changed then. 
For 2.: “_extends” wouldn’t need to be used at all. Then there should be a new property e.g. “protocolExecution” or “relatedExperiment” with “_linkedTypes” pointing to a “protocolExecution” (I assume as an object rather than an array then).

Reply:
Thanks for the quick feedback Ulrike. I am trying to achieve option 1, but I wasn't sure about the rules for "_extends" in terms of extending or restricting existing fields. One approach would be to define a parent type like "Activity" that could be extended by both ProtocolExecution and Computation. The question of how to extend core types definitely needs to be addressed more generally, though, as I think this will be a common need in openMINDS extensions (ephys, etc.)

Reply:
Yes, this was what I was afraid of :D Option 2 would have been easier. 

@apdavison, you are right. @lzehl this is something that needs to be discussed. With every additional extension, some existing schemas may need to be split to allow this "recycling mechanism" that `"_extends"` entails.

The easy fix for this specific case is as you suggested, @apdavison. Our "protocolExecution" schema would need to extend a new schema which holds the properties that will be identical for both "protocolExecution" and "computation" - everything except for input, output and protocol. 
However, "computation" would not be able to have the property "input" itself. This would need to be part of the 3 sub-schemas that extend "computation" since the "input" for "simulation" expects other linkedTypes than "visualization"/"dataAnalysis".  


Slightly unrelated: I would personally like to suggest to avoid using a schema name twice. I got confused here thinking that protocol referred to the core/protocol until I noticed that there was a computation/protocol. If it's just me that gets confused by this, please do ignore this suggestion ;)

Reply:
As discussed, we should support this for submodules of the central repository. I would like to suggest     
"_extends": "/core/research/protocolExecution.schema.tpl.json",